File: leetcode/easy/3062-winner-of-the-linked-list-game/3062-winner-of-the-linked-list-game.kt
/**
 * Example:
 * var li = ListNode(5)
 * var v = li.`val`
 * Definition for singly-linked list.
 **/

class ListNode(var `val`: Int) {
    var next: ListNode? = null
}


class Solution {
    fun gameResult(head: ListNode?): String {
        var evenCount = 0
        var oddCount = 0
        var cur = head

        while (cur != null) {
            val even = cur.`val`
            val odd = cur.next.`val`

            if (even > odd) {
                evenCount++
            } else if (odd > even) {
                oddCount++
            }

            cur = cur.next.next
        }

        if (evenCount == oddCount) {
            return "Tie"
        }

        return if (evenCount > oddCount) "Even" else "Odd"
    }
}

File: leetcode/easy/3062-winner-of-the-linked-list-game/3062-winner-of-the-linked-list-game.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def gameResult(self, head: Optional[ListNode]) -> str:
        evenPoint = 0
        oddPoint = 0

        while (head):
            even = head.val
            odd = head.next.val

            if (even > odd):
                evenPoint += 1
            elif (even < odd):
                oddPoint += 1

            head = head.next.next

        if (evenPoint == oddPoint):
            return "Tie"

        return "Even" if evenPoint > oddPoint else "Odd"
